gtsam/base/treeTraversal/parallelTraversalTasks.h

```diff
@@ -34,89 +34,79 @@ namespace gtsam {
 
       /* ************************************************************************* */
       template<typename NODE, typename DATA, typename VISITOR_PRE, typename VISITOR_POST>
-      class PreOrderTask : public tbb::task
+      class PreOrderTask
       {
       public:
         const boost::shared_ptr<NODE>& treeNode;
         boost::shared_ptr<DATA> myData;
         VISITOR_PRE& visitorPre;
         VISITOR_POST& visitorPost;
         int problemSizeThreshold;
+        tbb::task_group& tg;
         bool makeNewTasks;
 
-        bool isPostOrderPhase;
+        mutable bool isPostOrderPhase;
```

Comment:
Add a comment about this?

Reply:
I think this is pretty self-explanatory and the variable is aptly named in the first place.
`isPostOrderPhase` is a variable that we want to change even in a `const` function.

I've kept the same comments throughout the code as before so I believe it is sufficient.
I'll add a message in the commit tho.

Reply:
Easy for you to say, but what about the next student who comes and looks at this 2 years later?

A simple comment on what does mutable do and what's the variable used for should suffice. The amount of keystrokes you spent adding the comment above, you could have used in adding the code comment. :)

Plus, there's no such thing as too much documentation.

Reply:
Also, comments in commit messages help no one especially if the person who made those commits decides to move on to other things. GTSAM gets many commits so it is easy to lose track.

Reply:
sure i'll add something

Reply:
let me know if you think it is appropriate